File: mobile/android/core/designsystem/src/main/kotlin/com/slopssaloon/omen/core/designsystem/component/OmenCard.kt
package com.slopssaloon.omen.core.designsystem.component

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.semantics.stateDescription
import androidx.compose.ui.unit.dp
import androidx.compose.foundation.shape.RoundedCornerShape
import com.slopssaloon.omen.core.designsystem.theme.OmenTheme

/** Registry §3.1 Card / Surface variants. Cards are display containers; interactive rows use their own contract. */
enum class OmenCardVariant { Solid, Outlined, Empty, Error, Preview }

/** Semantic card emphasis — neutral foundation, Omen signal, or risk. */
enum class OmenCardTone { Neutral, Omen, Risk }

/**
 * Token-backed Material3 surface for Omen compositions. This intentionally has no click handler:
 * selection/focus behavior belongs to the approved interactive component that composes it, rather
 * than turning every display card into a misleading button.
 */
@Composable
fun OmenCard(
    modifier: Modifier = Modifier,
    variant: OmenCardVariant = OmenCardVariant.Solid,
    tone: OmenCardTone = OmenCardTone.Neutral,
    contentPadding: PaddingValues = PaddingValues(OmenTheme.spacing.cardInterior),
    content: @Composable () -> Unit,
) {
    val colors = OmenTheme.color
    val background = when (variant) {
        OmenCardVariant.Preview -> colors.surface2
        else -> colors.surface1
    }
    val toneBorder = when (tone) {
        OmenCardTone.Neutral -> if (variant == OmenCardVariant.Outlined) colors.border else colors.borderSubtle
        OmenCardTone.Omen -> colors.omen
        OmenCardTone.Risk -> colors.data.riskHigh
    }
    val borderColor = if (variant == OmenCardVariant.Error) colors.data.riskHigh else toneBorder

    Surface(
        modifier = modifier.then(
            if (variant == OmenCardVariant.Error) Modifier.semantics { stateDescription = "Error" } else Modifier,
        ),
        shape = RoundedCornerShape(12.dp),
        color = background,
        contentColor = colors.textPrimary,
        border = BorderStroke(1.dp, borderColor),
        tonalElevation = 0.dp,
        shadowElevation = 0.dp,
    ) { Box(Modifier.padding(contentPadding)) { content() } }
}

File: mobile/ios/OmenIOS/OmenIOS/DesignSystem/OmenCard.swift
import SwiftUI

/// Registry §3.1 Card / Surface variants. Cards are display containers by default.
enum OmenCardVariant { case solid, outlined, empty, error, preview }

enum OmenCardTone { case neutral, omen, risk }

/// Token-backed SwiftUI surface for Omen compositions. Interactive behavior belongs to the
/// approved component that composes this display container, not to Card itself.
struct OmenCard<Content: View>: View {
    var variant: OmenCardVariant = .solid
    var tone: OmenCardTone = .neutral
    var contentPadding: CGFloat = OmenSpacing.cardInterior
    @ViewBuilder let content: Content

    init(
        variant: OmenCardVariant = .solid,
        tone: OmenCardTone = .neutral,
        contentPadding: CGFloat = OmenSpacing.cardInterior,
        @ViewBuilder content: () -> Content
    ) {
        self.variant = variant
        self.tone = tone
        self.contentPadding = contentPadding
        self.content = content()
    }

    private var background: Color { variant == .preview ? OmenColor.surface2 : OmenColor.surface1 }

    private var border: Color {
        if variant == .error { return OmenColor.Data.riskHigh }
        switch tone {
        case .neutral: return variant == .outlined ? OmenColor.border : OmenColor.borderSubtle
        case .omen: return OmenColor.omen
        case .risk: return OmenColor.Data.riskHigh
        }
    }

    var body: some View {
        content
            .padding(contentPadding)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(background)
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .overlay(RoundedRectangle(cornerRadius: 12).stroke(border, lineWidth: 1))
            .accessibilityValue(variant == .error ? "Error" : "")
    }
}
